### PR TITLE
raft: port inflights test

### PR DESCRIPTION
**Pull request by cssivision**

### CODE REVIEW

File: src/progress.rs [hunk 1]
```diff
@@ -370,3 +370,139 @@ impl Inflights {
...
+
+        let mut inflight2 = Inflights{
+            start: 5,
+            buffer: Vec::with_capacity(10),
+            ..Default::default()
+        };
+        inflight2.buffer.extend_from_slice(&vec![0, 0, 0, 0, 0]);
+
+        for i in 0..5 {
+            println!("{}", i);
```
Comment (Hoverbear) [inline src/progress.rs]: Is this left in accidently?

Comment (cssivision) [reply src/progress.rs]: fixed

Comment (Hoverbear) [reply src/progress.rs]: Thanks!

**Comment by siddontang** [comment]
PTAL @BusyJay

**Pull request merged**